mitype/app.py

```diff
@@ -161,8 +161,11 @@ def initialize(self, win):
         )
 
         # If required number of lines are more than the window height, exit
-        # +3 for printing stats at the end of the test
-        if self.line_count + 3 > self.window_height:
+        # +5 for printing stats at the end of the test
+        if (
+            self.line_count + 5 >= self.window_height
+            or self.line_count >= self.window_width
+        ):
             curses.endwin()
             sys.stdout.write("Window too small to print given text")
             sys.exit(1)
```

Comment:
This code block is being repeated twice. So let's move it to its own separate function. And then call the function here.

Reply:
Right!